koku/koku/trino_database.py

```diff
@@ -149,6 +148,15 @@ def executescript(trino_conn, sqlscript, *, params=None, preprocessor=None):
                 cur.execute(stmt, params=s_params)
                 results = cur.fetchall()
             except TrinoQueryError as trino_exc:
+                if "NoSuchKey" in str(trino_exc) and trino_external_error_retries > 0:
+                    LOG.debug("TrinoExternalError Exception, retrying...")
+                    return executescript(
+                        trino_conn=trino_conn,
+                        sqlscript=sqlscript,
+                        params=params,
+                        preprocessor=preprocessor,
+                        trino_external_error_retries=trino_external_error_retries - 1,
+                    )
```

Comment:
This should probably include delay and/or backoff. Adding some jitter may also be helpful to prevent multiple processes from retrying at the same time.